RoboJackets/robocup-software
# Planner node considers ball an obstacle

## Description

This PR updates the planning system to consider the ball a dynamic obstacle, and plan paths that keep a certain distance away from the ball as published by the coach node

## Associated / Resolved Issue
Resolves [make planner respect ball on stop](https://app.clickup.com/t/34c4duz)

## Design Documents
[Link](link-to-design-doc)

## Steps to Test
### Test Case 1
1. Run the sim (green check/normal mode). Currently, six robots move in a predefined manner. Robot 1 (the defender) is the only one set to pay any attention to the ball. 
2. Place the ball in/near the path of robot 1

**Expected result:** Robot 1 touching the ball

### Test Case 2: STOP
1. Run the sim in STOP. (Coach node should be publishing a min_dist_from_ball at 0.5)
2. Place the ball in/near the path of robot 1

**Expected result:** Robot 1 should keep a larger radius (0.5 in some unit?) around the ball.

## Key Files to Review
Group 1
 * File 1
 * File 2

Group 2
 * File 3
 * File 4

## Review Checklist


## (Optional) Sub-issues (for drafts)
_Note: if you find yourself breaking this PR into many smaller features, it may make sense to break up the PR into logical units based on these features._
